lib/internal/util.js

```diff
@@ -187,3 +191,65 @@ exports.convertToValidSignal = function convertToValidSignal(signal) {
 
   throw new Error('Unknown signal: ' + signal);
 };
+
+const kCustomPromisifiedSymbol = Symbol('util.promisify.custom');
+const kCustomPromisifyArgsSymbol = Symbol('customPromisifyArgs');
+
+function promisify(orig) {
+  if (typeof orig !== 'function') {
+    throw new TypeError('The first argument to promisify() must be a function');
+  }
+
+  if (orig[kCustomPromisifiedSymbol]) {
+    const fn = orig[kCustomPromisifiedSymbol];
```

Comment:
I'm not worried about performance - just visibility. `String(Object.getOwnPropertySymbols(util.promisify(() => {}))[0])) === 'Symbol(util.promisify.custom)` would become part of the API, or at least something to debate about changing later; Code relying on `Object.getOwnPropertySymbols(func)` could suddenly break if someone calls `util.promisify` on it (which means it's susceptible to any mutation, but that doesn't mean node core should be the cause of it); if you later want to cache more information than the function itself, the implementation of this caching suddenly could become semver-major; `const sym = Object.getOwnPropertySymbols(util.promisify(() => {}))[0]; func[sym] = notAFunction;` would mean that `util.promisify(func)` would not reliably produce a function that returns a promise (which absolutely could break code)…

Is the purpose of the symbol so that users don't need to call `util.promisify` later, or is the purpose caching? If the former, what's the utility? If the latter, why not use an implementation detail that's invisible to consumers?